src/analyze/analyze-chid.c

```diff
@@ -223,6 +252,65 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(sd_device *drm_card, char16_t **ret_panel) {
+        assert(drm_card);
+        assert(ret_panel);
+        int r;
+        const char *edid_content = NULL;
+        size_t edid_size = 0;
+        r = sd_device_get_sysattr_value_full(drm_card, "edid", &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+
+        EdidHeader header;
+        if (!edid_parse_blob(edid_content, edid_size, &header))
+                return -EINVAL;
+
+        *ret_panel = new0(char16_t, 8);
```

Comment:
our userspace allocations can fail. and we have this rule that return params should never be touched on error, and definitely be initialized on success. here you touch `ret_panel` before the last call that can fail.

hence please allocate memory here, in a temporary variable, and once everything checked out move it into `ret_panel` with TAKE_PTR()